exasol_transformers_extension/udfs/sequence_classification_single_text_udf.py

```diff
@@ -16,12 +16,19 @@ def __init__(self,
         self.bacth_size = batch_size
         self.base_model = base_model
         self.tokenizer = tokenizer
+        self.device = None
         self.cache_dir = None
         self.last_loaded_model_name = None
         self.last_loaded_model = None
         self.last_loaded_tokenizer = None
 
     def run(self, ctx):
+        device_id = ctx.get_dataframe(1).iloc[0]['device_id']
+        device_name = f"cuda:{device_id}" \
+            if torch.cuda.is_available() and device_id is not None else "cpu"
+        self.device = torch.device(device_name)
+        ctx.reset()
+
         while True:
             batch_df = ctx.get_dataframe(self.bacth_size)
```

Comment:
if the device is the first parameter, you use ctx.get_dataframe(self.batch_size, start_col=1)
https://github.com/exasol/script-languages-release/blob/master/doc/user_guide/py_dataframe.md#accessing-data